README.md

```diff
@@ -30,20 +30,38 @@ Edit this document to include your answers after each question. Make sure to lea
 
 1. Describe the biggest difference between `.forEach` & `.map`.
 
+.map() allocates memory and stores return values. .forEach() throws away return values and always returns undefined. 
+.forEach() will allow a callback function to mutate the current array. map() will instead return a new array.
+
+
 2. What is the difference between a function and a method?
 
+‘Method’ is used to describe functions that are properties of an object that isn’t a global one.
+
 3. What is closure?
 
+A closure is a function having access to the parent scope, even after the parent function has closed.
+
 4. Describe the four rules of the 'this' keyword.
 
+1 - Window Binding: 'this' points at the window/console object when in the global scope.
+2 - Implicit Binding: Whenever a function is called by a preceding dot, the object before that dot is 'this'.
+3 - New Binding: Whenever a constructor function is used, this refers to the specific instance of the object that is created and returned by the constructor function.
+4 - Explicit binding: Whenever JavaScript’s call or apply method is used, this is explicitly defined.
+
 5. Why do we need super() in an extended class?
 
+Because constructors in inheriting classes must call super() before using ‘this’ to create new attributes. Constructor functions of an inheriting classes are labelled with a special internal property [[ConstructorKind]]:"derived".
+
+When a normal constructor runs, it creates an empty object as ‘this’ and continues with it. But when a derived constructor runs, it doesn’t do it. It expects the parent constructor to do this job. So if we’re making a constructor of our own, then we must call super, because otherwise the object with 'this' reference to it won’t be created. And we’ll get an error.
+
+
 ### Git Set up
 
-* [ ] Fork the project into your GitHub user account
-* [ ] Clone the forked project into a directory on your machine
-* [ ] Create a pull request before you start working on the project requirements.  You will continuously push your updates throughout the project.
-* [ ] You are now ready to build this project with your preferred IDE
+* [X] Fork the project into your GitHub user account
+* [X] Clone the forked project into a directory on your machine
+* [X] Create a pull request before you start working on the project requirements.  You will continuously push your updates throughout the project.
+* [X] You are now ready to build this project with your preferred IDE
 
 ## Minimum Viable Product
 
```

Comment:
Straight to the point without over-explaining. Really great job on these answers, Sean!